examples/simple/simple.cpp

```diff
@@ -6,11 +6,49 @@
 #include <string>
 #include <vector>
 
+// a function that can be called for every computed node during graph evaluation
+// the user can choose to whether to observe the data of the node depending on the tensor parameters
+static bool observe_compute(struct ggml_tensor * t, bool ask, void * user_data) {
+    GGML_UNUSED(user_data);
+
+    // the scheduler is asking us if we want to observe this node
+    if (ask) {
+        // check if name contains soft_max (customize to your needs)
+        return strstr(t->name, "soft_max") != 0;
+    }
+
+    // print the node info
+    printf("%s: t->name = %32s, t->op = %12s, [%5d, %5d, %5d, %5d]\n",
+            __func__, t->name, ggml_op_name(t->op), (int) t->ne[0], (int) t->ne[1], (int) t->ne[2], (int) t->ne[3]);
+
+    // this will copy the data to host memory (if needed)
+    static std::vector<float> t_data;
+
+    const bool is_host = ggml_backend_buffer_is_host(t->buffer);
+
+    if (!is_host || !ggml_is_contiguous(t)) {
+        t_data.resize(ggml_nelements(t));
+        ggml_backend_tensor_get(t, t_data.data(), 0, ggml_nbytes(t));
+    }
+
+    const float * data = is_host ? (const float *) t->data : t_data.data();
```

Comment:
I don't understand the use if `ggml_is_contiguous` here. `ggml_backend_tensor_get` does not make the data contiguous, and the result is ignored.